src/components/general/CanvasDrawModal/index.tsx

```diff
@@ -0,0 +1,271 @@
+            const bcr = canvasRef.current.getBoundingClientRect();
+
+            startRef.current = {
+                x: e.clientX - bcr.left,
+                y: e.clientY - bcr.top,
+            };
+
+            endRef.current = {
+                ...startRef.current,
+            };
+
+            onDraw(startRef.current, endRef.current);
+        }
+    }, [canvasRef, onDraw]);
+
+    const handleMouseUp = React.useCallback(() => {
+        mouseDownRef.current = false;
+    }, []);
+
+    const handleMouseMove = React.useCallback((e: MouseEvent) => {
+        if (mouseDownRef.current && canvasRef.current) {
+            const bcr = canvasRef.current.getBoundingClientRect();
+
+            endRef.current = {
+                x: e.clientX - bcr.left,
+                y: e.clientY - bcr.top,
+            };
+
+            onDraw(startRef.current, endRef.current);
+
+            startRef.current = {
+                ...endRef.current,
+            };
+        }
+    }, [onDraw, canvasRef]);
+
+    React.useEffect(() => {
+        const canvas = canvasRef.current;
+
+        if (canvas) {
+            canvas.addEventListener('mousedown', handleMouseDown);
+            document.addEventListener('mouseup', handleMouseUp);
+            canvas.addEventListener('mousemove', handleMouseMove);
+        }
+
+        return () => {
+            if (canvas) {
+                canvas.removeEventListener('mousedown', handleMouseDown);
+                document.removeEventListener('mouseup', handleMouseUp);
+                canvas.removeEventListener('mousemove', handleMouseMove);
+            }
+        };
+    }, [handleMouseDown, handleMouseUp, handleMouseMove, canvasRef]);
+}
+
+const penSizeOptions = [
+    { key: 1, label: _ts('components.canvasDrawModal', 'penSizeOptionSmallLabel') },
+    { key: 3, label: _ts('components.canvasDrawModal', 'penSizeOptionMediumLabel') },
+    { key: 7, label: _ts('components.canvasDrawModal', 'penSizeOptionLargeLabel') },
+];
+const DEFAULT_PEN_SIZE = penSizeOptions[1].key;
+
+const drawImage = (
+    canvasRef: React.RefObject<HTMLCanvasElement>,
+    img: HTMLImageElement,
+    scale: number,
+) => {
+    if (canvasRef.current) {
+        const context = canvasRef.current.getContext('2d');
+        if (context) {
+            context.drawImage(
+                img,
+                0, 0, img.width * scale, img.height * scale,
+            );
+        }
+    }
+};
+
+interface Props {
+    className?: string;
+    imgSrc?: string;
+    onDone?: (imgData: string) => void;
+    onCancel?: () => void;
+}
+
+function CanvasDraw(props: Props) {
+    const {
+        className,
+        imgSrc,
+        onDone,
+        onCancel,
+    } = props;
+    const canvasRef = React.useRef<HTMLCanvasElement>(null);
+    const canvasContainerRef = React.useRef<HTMLDivElement>(null);
+    const [color, setColor] = React.useState(
+        getComputedStyle(document.body).getPropertyValue('--color-accent'),
```

Comment:
Use `useState(() => the value to be computed)` syntax

Reply:
but why?

Reply:
Because, the `getComputedStyle(document.body).getPropertyValue('--color-accent')` will run on every render.
We can avoid that using this syntax:
```suggestion
        () => getComputedStyle(document.body).getPropertyValue('--color-accent'),
```